node/network/collator-protocol/src/validator_side.rs

```diff
@@ -48,6 +49,47 @@ const COST_REQUEST_TIMED_OUT: Rep = Rep::new(-20, "A collation request has timed
 const COST_REPORT_BAD: Rep = Rep::new(-50, "A collator was reported by another subsystem");
 const BENEFIT_NOTIFY_GOOD: Rep = Rep::new(50, "A collator was noted good by another subsystem");
 
+#[derive(Clone, Default)]
+pub(super) struct Metrics(Option<MetricsInner>);
+
+impl Metrics {
+	fn on_request(&self, succeeded: bool) {
```

Comment:
maybe it's worth changing the type to `succeeded: Result<(), ()>` for clarity, but it's more verbose as well.